src/tracer.cpp

```diff
@@ -8,14 +8,24 @@ static std::shared_ptr<Tracer>& get_global_tracer() {
   return global_tracer;
 }
 
+static std::atomic<bool>& get_is_global_tracer_registered() {
+  static std::atomic<bool> is_global_tracer_registered{false};;
+  return is_global_tracer_registered;
+}
+
 std::shared_ptr<Tracer> Tracer::Global() noexcept {
   return get_global_tracer();
 }
 
 std::shared_ptr<Tracer> Tracer::InitGlobal(
     std::shared_ptr<Tracer> tracer) noexcept {
   get_global_tracer().swap(tracer);
+  get_is_global_tracer_registered().store(true);
```

Comment:
There's a race condition here if IsGlobalTracerRegistered gets called between lines 22 and 23.